src/main/java/com/google/sps/servlets/ProfileRoutes.java

```diff
@@ -30,13 +30,23 @@
 @SuppressWarnings("serial")
 @WebServlet("/user-routes")
 public class ProfileRoutes extends HttpServlet {
+  static class RouteWithType {
```

Comment:
I don't think it is neccessary to create a new class to store a single extra property. It would be simpler to add the property to the Route class, and just use a default value (perhaps ReadOnly) when it is not specified. The data class does not have to match the database exactly, as this field will not be stored with the rest of the data.

If you do add a new class, it would be good to add javadoc comments to explain what it is used for.